extensions/resteasy-reactive/rest-client-reactive/runtime/src/main/java/io/quarkus/rest/client/reactive/runtime/ConfigUtils.java

```diff
@@ -9,20 +9,51 @@ public class ConfigUtils {
 
     private static final Logger log = Logger.getLogger(ConfigUtils.class);
 
+    static final String PREFIX = "${";
+    static final String SUFFIX = "}";
+
+    /**
+     * Interpolates the given expression. The expression is expected to be in the form of ${config.property.name}.
+     *
+     * @param expression the expression to interpolate
+     * @param required whether the expression is required to be present in the configuration
+     * @return null if the resulting expression is empty, otherwise the interpolated expression
+     */
+    public static String interpolate(String expression, boolean required) {
+        StringBuilder sb = new StringBuilder(expression);
+        int idx;
+        while ((idx = sb.lastIndexOf(PREFIX)) > -1) {
+            int endIdx = sb.indexOf(SUFFIX, idx);
+            if (endIdx < 0) {
+                throw new IllegalArgumentException("Invalid expression: " + expression);
+            }
+            String configValue = getConfigValue(sb.substring(idx, endIdx + 1), required);
+            // If no value is found, we return null directly
+            if (configValue == null) {
+                return null;
+            }
```

Comment:
@geoand I added this after the last review. The idea is that if the config property is not found, the expression itself is null (not used). 

Eg. `Bearer ${property}` - if `${property}` is not set then the header parameter is not set

Reply:
👍